src/Test/StateMachine/Parallel.hs

```diff
@@ -238,18 +239,20 @@ generateNParallelCommands sm@StateMachine { initModel } np =
 
 -- | A list of commands is parallel safe if the pre-conditions for all commands
 --   hold in all permutations of the list.
-parallelSafe :: StateMachine model cmd m resp -> model Symbolic
+parallelSafe :: Rank2.Foldable resp
+             => StateMachine model cmd m resp -> model Symbolic
              -> Commands cmd resp -> Bool
-parallelSafe StateMachine { precondition, transition } model0
+parallelSafe StateMachine { precondition, transition, mock } model0
   = and
   . map (preconditionsHold model0)
   . permutations
   . unCommands
   where
-    preconditionsHold _     []                              = True
-    preconditionsHold model (Command cmd resp _vars : cmds) =
+    preconditionsHold _     []                             = True
+    preconditionsHold model (Command cmd resp vars : cmds) =
         boolean (precondition model cmd) &&
-          preconditionsHold (transition model cmd resp) cmds
+          preconditionsHold (transition model cmd resp) cmds &&
+          length vars == length (getUsedVars $ fst $ runGenSym (mock model cmd) newCounter)
```

Comment:
Could you please add a comment here about why this is needed and maybe also link to this PR?

Reply:
> Could you please add a comment here about why this is needed and maybe also link to this PR?

Added comment and rebased to new master